nautobot/dcim/views.py

```diff
@@ -4139,9 +4139,12 @@ def get_extra_context(self, request, instance):
             context["device_type_table"] = device_type_table
 
             total_devices = 0
+            device_type_count = 0
             for device_type in device_types:
                 total_devices += device_type.device_count
+                device_type_count += 1
             context["total_devices"] = total_devices
+            context["device_type_count"] = device_type_count
```

Comment:
```suggestion
            for device_type in device_types:
                total_devices += device_type.device_count
            context["total_devices"] = total_devices
            context["device_type_count"] = device_types.count()
```

Reply:
I thought about this, but it seemed like an extra query

Reply:
Fair point - would be good to test either way. Could also use `len(device_types)` as that shouldn't be an extra query.